crates/devolutions-pedm/src/api/mod.rs

```diff
@@ -155,34 +160,175 @@ pub fn openapi() -> OpenApi {
     api
 }
 
-pub(crate) async fn serve(pipe_name: &'static str) -> Result<()> {
-    let app = api_router();
+/// A handler that is not part of the public API.
+async fn health_check() -> &'static str {
+    "OK"
+}
+
+pub async fn serve(pipe_name: &'static str, config_path: Option<Utf8PathBuf>) -> Result<(), ServeError> {
+    let state = AppState::load(config_path).await?;
+
+    // a plain Axum router
+    let hello_router = Router::new().route("/health", axum::routing::get(health_check));
+
+    let app = api_router()
+        .merge(ApiRouter::from(hello_router))
+        .layer(LogLayer::new(state.clone())) // apply to the merged router
+        .with_state(state.clone());
 
     let mut make_service = app.into_make_service_with_connect_info::<RawNamedPipeConnectInfo>();
 
     let mut server = create_pipe(pipe_name)?;
+
+    // log the server startup
+    let run_id = state.db.log_server_startup(pipe_name).await?;
+    info!("Started server at {pipe_name} with run ID {run_id}");
+    info!(
+        "Starting request ID counter at {}",
+        state.req_counter.load(Ordering::Relaxed)
+    );
+
     loop {
         server.connect().await?;
         let client = server;
 
         server = create_pipe(pipe_name)?;
 
-        let tower_service = make_service.call(&client).await?;
+        let tower_service = make_service.call(&client).await.unwrap(); // return type is Infallible
```

Comment:
style: A better way of handling Infaillible is:

```suggestion
        let Ok(tower_service) = make_service.call(&client).await;
```